src/p2p/potato_udp.erl

```diff
@@ -0,0 +1,58 @@
+-module(potato_udp).
+-behavior(gen_server).
+
+-export([init/1, handle_call/3, handle_cast/2, handle_info/2, code_change/3, terminate/2]).
+
+%% state maps {game_id} to game instance PID
+-spec init(integer()) -> {ok, {gen_udp:socket(), map()}}.
+init(Port) ->
+  io:format("gen_udp open on port: ~p~n", [Port]),
+  {ok, Socket} = gen_udp:open(Port, [binary, {active,true}]),
+  %% add to group?
+  {ok, {Socket, maps:new()}}.
+
+%% Never used
+handle_call(_E, _From, S) ->
+  {noreply, S}.
+
+%% add a {game_id} => Pid to map
+handle_cast({add_game,Key,Pid}, S) ->
+  {Socket, Map} = S,
+  Map2 = maps:put(Key, Pid, Map),
+  {noreply, {Socket, Map2}};
+
```

Comment:
you can also do `{add_game,Key,Pid}, {Socket, Map})` instead of `S`

Reply:
:O